src/orion/core/worker/primary_algo.py

```diff
@@ -21,39 +33,67 @@ class SpaceTransformAlgoWrapper:
 
     Parameters
     ----------
-    algo_constructor: Child class of `BaseAlgorithm`
-        Class constructor to build the algorithm object.
+    algorithm: instance of `BaseAlgorithm`
+        Algorithm to be wrapped.
     space : `orion.algo.space.Space`
        The original definition of a problem's parameters space.
     algorithm_config : dict
        Configuration for the algorithm.
 
     """
 
-    def __init__(self, algo_constructor, space, **algorithm_config):
+    def __init__(self, algorithm: AlgoType, space: Space):
         self._space = space
-        requirements = backward.get_algo_requirements(algo_constructor)
-        self.transformed_space = build_required_space(space, **requirements)
-        self.algorithm = algo_constructor(space, **algorithm_config)
-        self.algorithm.space = self.transformed_space
+        self.algorithm: AlgoType = algorithm
+        self.registry = Registry()
+        self.registry_mapping = RegistryMapping(
+            original_registry=self.registry,
+            transformed_registry=self.algorithm.registry,
```

Comment:
Right, the transformed_registry is the one from the algorithm. Because of this it is better indeed that mapping does not register for the registers otherwise the algorithm would need to account for the fact that someone externally may mess with its registry...

Reply:
Yep :slightly_smiling_face: